### ISSUE TITLE
Warpctc gradients != torch.nn gradients

### ISSUE THREAD

**Issue by swethmandava** [SeanNaren/deepspeech.pytorch#364]
from torch.nn import CTCLoss
from warpctc_pytorch import CTCLoss as ctcloss2
criterion = CTCLoss(reduction="none")
warpctc_criterion = ctcloss2()

nn_loss = criterion(softmax_out, targets, output_seq_len, target_sizes)
nn_loss = nn_loss.sum()
optimizer.zero_grad()
nn_loss.backward()
nn_norm = get_norm(model.parameters())

warpctc_loss = warpctc_criterion(linear_out, targets, output_seq_len, target_sizes)
optimizer.zero_grad()
warpctc_loss.backward()
warpctc_norm = get_norm(model.parameters())

When I try this, warpctc_loss = nn_loss but warpctc_norm != ctc_norm. I'm not scaling the loss, I know there's a bug wrt that in warpctc, is there anything else that might cause this?

**Comment by swethmandava** [comment]
Update :
If I disable cuda, then the norms match. Error is with torch.nn , not warpctc!

**Comment by SeanNaren** [comment]
Try using the `pytorch-nightly` build! Let me know if that helps (`conda install pytorch-nightly`)
